core/src/smartcontracts/isi/permissions.rs

```diff
@@ -525,63 +1145,113 @@ impl<W: WorldTrait, O: NeedsPermission> IsAllowed<W, O> for AllowAll {
         _authority: &AccountId,
         _instruction: &O,
         _wsv: &WorldStateView<W>,
-    ) -> Result<(), DenialReason> {
+    ) -> Result<()> {
         Ok(())
     }
 }
 
+impl From<AllowAll> for IsInstructionAllowedBoxed {
```

Comment:
These could (and probably should) be implemented in a declarative macro. Suggested syntax
```rust
impl_from_for_allowed_boxed! {
     AllowAll => IsInstructionAllowedBoxed,
     AllowAll => IsQueryAllowedBoxed,
     AllowAll => IsExpressionAllowedBoxed,
}